File: mycollab-web/src/main/java/com/esofthead/mycollab/module/mail/service/SendingRelayEmailNotificationAction.java
package com.esofthead.mycollab.module.mail.service;

import com.esofthead.mycollab.common.domain.SimpleRelayEmailNotification;

public interface SendingRelayEmailNotificationAction {
	void sendNotificationForCreateAction(
			SimpleRelayEmailNotification notification);

	void sendNotificationForUpdateAction(
			SimpleRelayEmailNotification notification);

	void sendNotificationForCommentAction(
			SimpleRelayEmailNotification notification);
}

File: mycollab-services/src/main/scala/com/mycollab/schedule/email/SendingRelayEmailNotificationAction.scala
package com.mycollab.schedule.email

import com.mycollab.common.domain.SimpleRelayEmailNotification

/**
  * @author MyCollab Ltd
  * @since 5.1.0
  */
trait SendingRelayEmailNotificationAction {
  def sendNotificationForCreateAction(notification: SimpleRelayEmailNotification): Unit

  def sendNotificationForUpdateAction(notification: SimpleRelayEmailNotification): Unit

  def sendNotificationForCommentAction(notification: SimpleRelayEmailNotification): Unit
}
